cdci_data_analysis/configurer.py

```diff
@@ -33,89 +33,62 @@
-
-        if data_server_port is not None:
-            logger.warning(
-                "data_server_port is disregarded, since it is naturally included in the url")
-
-        logger.info("building config from %s %s %s", data_server_url,
-                    data_server_remote_cache, dispatcher_mnt_point)
+    def __init__(self, **kwargs):
 
-        if data_server_url is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_url is None: {data_server_url}")
+        conf = kwargs.copy()
 
-        if data_server_remote_cache is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_remote_cache is None: {data_server_remote_cache}")
+        logger.info("building config from %s", conf)
 
-        if dispatcher_mnt_point is None:
-            logger.warning(
-                f"problem constructing {self}: dispatcher_mnt_point is None: {dispatcher_mnt_point}")
+        required_keys = ['data_server_url']
 
-        self.data_server_url = data_server_url
+        for key in required_keys:
+            try:
+                self.__setattr__(key, conf.pop(key))
+            except KeyError as e:
+                logger.error(
+                    f"problem constructing {self}: {key} configuration key is required")
+                raise e
 
-        # dummy prods local cache
-        self.dummy_cache = dummy_cache
+        if conf.pop('data_server_port', None) is not None:
+             logger.warning(
+                 "data""_server_port is disregarded, since it is naturally included in the url")
 
-        # path to dataserver cache
-        self.data_server_remote_path = data_server_remote_cache
+        #special cases (maybe INTEGRAL specific a bit)
+        if conf.get('data_server_remote_cache', None) is not None:
+            # path to dataserver cache
+            self.data_server_remote_path = conf.pop('data_server_remote_cache')
+        else:
+            self.data_server_remote_path = None
 
-        if dispatcher_mnt_point is not None:
-            self.dispatcher_mnt_point = os.path.abspath(dispatcher_mnt_point)
+        if conf.get('dispatcher_mnt_point', None) is not None:
+            self.dispatcher_mnt_point = os.path.abspath(conf.pop('dispatcher_mnt_point'))
             FilePath(file_dir=self.dispatcher_mnt_point).mkdir()
         else:
             self.dispatcher_mnt_point = None
 
-        self.products_url = products_url
-
         if self.dispatcher_mnt_point is not None and self.data_server_remote_path is not None:
-            #self.data_server_cache = os.path.join(self.dispatcher_mnt_point, self.data_server_remote_path)
             self.data_server_cache = os.path.join(
                 self.dispatcher_mnt_point, self.data_server_remote_path)
         else:
             self.data_server_cache = None
 
+        #optional config keys
+        for key in conf:
+            self.__setattr__(key, conf[key])
```

Comment:
To preserve the compactness of the code, we could define, as indeed suggested, a place for each instrument (plugin seems indeed the best place) where the instrument-specific keywords are defined.

And we could use the same approach for those mandatory ones, simply placing those somewhere else in a dispatcher configuration.

Then, we could separate the loading of those two sets of keys, (eg general and instrument-specific), and treat them in separate ways.

Reply:
indeed!

there are two key classifications: required/optional and dispatcher/plugin. 2x2=4 options in total. 
But I do not think there should be any key which is not validated by some of these lists.

It is probably a healthy practice if typos, mistakes, misunderstandings in the config  are noticed.

We could even use config schema (e.g. with python schema-config, which addresses other concerns, e.g. secret and templating handling, or similar) instead of defining programmatically.
But at this point it would be easier not to go into that, unless someone is curious. But we might want to finish this request.

At this point, we can just classify currently set values as follows:

**required, common**
data_server_url 

**required in production**
dummy_cache
products_url

**required, integral, special treatment**
dispatcher_mnt_point
data_server_remote_cache

**to discard**:
data_server_port

everything else may be validated with extra, optional (to preserve interface) lists provided by the plugin